src/lib/widgets/now-playing/specs/swipe-down-bar.spec.ts

```diff
@@ -0,0 +1,87 @@
+import svelte from 'svelte-inline-compile';
+import { fireEvent, render, screen } from '@testing-library/svelte';
+import { createEvent } from '@testing-library/dom';
+import { noop } from 'lodash-es';
+
+import { useLocalVars } from '$lib/shared/lib/jest-hacks';
+import SwipeDownBar from '../ui/swipe-down-bar.svelte';
+
+it('closes on the swipe', async () => {
+  const mockCloseHandler = jest.fn();
+
+  const swipeDownBar = useLocalVars(svelte`<SwipeDownBar on:minimize={mockCloseHandler} />`, [
+    SwipeDownBar,
+    mockCloseHandler,
+  ]);
+  render(swipeDownBar);
+
+  /**
+   * This logic is a mix of Testing Library `createEvent` function and an article
+   * https://www.freecodecamp.org/news/how-to-write-better-tests-for-drag-and-drop-operations-in-the-browser-f9a131f0b281/.
+   *
+   * Basically, it is a normal practice to test drag-and-drop (or pointerdown and pointermove) features
+   * by firings proper events with corresponding data (in this case, clientY property), but Testing
+   * Library's `createEvent` cannot assign the properties to Event entity. That's why the `Object.assign` is used.
+   */
+  fireEvent(
+    screen.getByRole('button'),
+    Object.assign(
```

Comment:
praise: this is some intense logic there, and I appreciate the research

issue: However, from what I discovered about JSDOM and the Testing Library is that testing drag is not a good idea in JSDOM, since it doesn't have layout at all. Maintainers of Testing Library suggest using Cypress for testing such stuff since it requires a real browser.

suggestion: perhaps we shouldn't cover this particular component with unit tests from Jest?

References:
- https://github.com/testing-library/user-event/issues/440#issuecomment-685010755 (This Kent C. Dodds guy is probably the most respectable guy in terms of testing from my current perspective)
- https://github.com/testing-library/user-event/issues/871 (An issue I created for the Testing Library when I was testing the scrubbing bar)

Reply:
Yeah, I've read these too, but couldn't understand what "no layout" actually mean. I'd appreciate you explaining the concept, and will easily remove the test

Reply:
Well, in simple terms, there's no "coordinate system", elements don't actually have a width and height. Everything is just collapsed into a single dot. That's how I understand it, anyway

Reply:
Which is why writing code that involves distances is risky and flaky in JSDOM

Reply:
Removed the test